src/regressors/linear_regression.cpp

```diff
@@ -3,6 +3,10 @@
 
 namespace cpp_ml::models::regressors {
 
+linear_regression :: linear_regression ()
+	:degree_(1)
```

Comment:
Here instead of setting directly `degree_`, you should call the polynomial_regression constructor:

`:polynomial_regression(1)`